### PR TITLE
[Dist] decouple num_chunks and num_parts for graphs with edge feature

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

This PR includes fix for https://github.com/dmlc/dgl/issues/4658 and https://github.com/dmlc/dgl/issues/4681. 
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tests/tools/create_chunked_dataset.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 def create_chunked_dataset(
-    root_dir, num_chunks, include_masks=False, include_edge_data=False
+    root_dir, num_chunks, include_masks=False
```
Comment (jermainewang) [inline tests/tools/create_chunked_dataset.py:18]: Maybe keep this option so you can test two cases: w/ and w/o edge data

**Review by jermainewang — APPROVED** [review]
LGTM

File: tests/tools/create_chunked_dataset.py (hunk 1 above)
Comment (Rhett-Ying) [reply tests/tools/create_chunked_dataset.py:18]: this is a temporary argument that was added on purpose to bypass possible test failure caused by `decouple of num_chunks and num_parts` not supported for edge feature yet. It's ok to remove it.

Comment (jermainewang) [reply tests/tools/create_chunked_dataset.py:18]: What I mean is to test two cases (w/ and w/o edge data) separately to ensure that there is no bug when the data does not have any edge data. You can ignore me if you've already tested it.

Comment (Rhett-Ying) [reply tests/tools/create_chunked_dataset.py:18]: I have tested it...

**Pull request merged**